File: src/main/java/com/paragon/client/systems/module/hud/impl/Direction.java
package com.paragon.client.systems.module.hud.impl;

import com.paragon.api.util.player.PlayerUtil;
import com.paragon.api.util.string.EnumFormatter;
import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.impl.client.Colours;
import net.minecraft.util.text.TextFormatting;

public class Direction extends HUDModule {

    public static Direction INSTANCE;

    public Direction() {
        super("Direction", "Displays what direction you are facing");

        INSTANCE = this;
    }

    @Override
    public void render() {
        renderText("Direction " + TextFormatting.WHITE + EnumFormatter.getFormattedText(PlayerUtil.getDirection()) + " [" + PlayerUtil.getAxis(PlayerUtil.getDirection()) + "]", getX(), getY(), Colours.mainColour.getValue().getRGB());
    }

    @Override
    public float getWidth() {
        return getStringWidth("Direction " + EnumFormatter.getFormattedText(PlayerUtil.getDirection()) + " [" + PlayerUtil.getAxis(PlayerUtil.getDirection()) + "]");
    }

    @Override
    public float getHeight() {
        return getFontHeight();
    }
}

File: src/main/kotlin/com/paragon/client/systems/module/hud/impl/Direction.kt
@file:Suppress("SuspiciousVarProperty")

package com.paragon.client.systems.module.hud.impl

import com.paragon.api.util.player.PlayerUtil
import com.paragon.api.util.render.font.FontUtil
import com.paragon.api.util.string.StringUtil
import com.paragon.client.systems.module.hud.HUDModule
import com.paragon.client.systems.module.impl.client.Colours
import net.minecraft.util.text.TextFormatting

/**
 * @author Surge
 */
object Direction : HUDModule("Direction", "Displays what direction you are facing") {

    override fun render() {
        FontUtil.drawStringWithShadow(
            "Direction " + TextFormatting.WHITE + StringUtil.getFormattedText(PlayerUtil.direction) + " [" + PlayerUtil.getAxis(PlayerUtil.direction) + "]",
            x,
            y,
            Colours.mainColour.value.rgb
        )
    }

    override var width: Float = 0.0F
        get() = FontUtil.getStringWidth(
            "Direction " + StringUtil.getFormattedText(PlayerUtil.direction) + " [" + PlayerUtil.getAxis(
                PlayerUtil.direction
            ) + "]"
        )

    override var height = FontUtil.getHeight()
        get() = FontUtil.getHeight()

}